sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql

```diff
@@ -0,0 +1,410 @@
+  USING
+    (client_id)
+  LEFT JOIN
+    metrics_ping AS metrics
+  USING
+    (client_id)
+),
+first_seen_with_pings_new_data AS (
+  SELECT
+    first_seen.client_id,
+    first_seen.first_seen_date,
+    DATE(first_session.first_run_datetime) AS first_run_date,
+    first_session.first_reported_country AS first_reported_country,
+    first_session.first_reported_isp AS first_reported_isp,
+    first_session.channel AS channel,
+    first_session.device_manufacturer AS device_manufacturer,
+    first_session.device_model AS device_model,
+    first_session.os_version AS os_version,
+    first_session.adjust_campaign AS adjust_campaign,
+    first_session.adjust_ad_group AS adjust_ad_group,
+    first_session.adjust_creative AS adjust_creative,
+    COALESCE(first_session.adjust_network, metrics.adjust_network) AS adjust_network,
+    metrics.install_source AS install_source,
+    STRUCT(
+      CASE
+      WHEN
+        first_session.client_id IS NULL
+      THEN
+        FALSE
+      ELSE
+        TRUE
+      END
+      AS reported_first_session_ping,
+      CASE
+      WHEN
+        metrics.client_id IS NULL
+      THEN
+        FALSE
+      ELSE
+        TRUE
+      END
+      AS reported_metrics_ping,
+      DATE(first_session.first_run_datetime) AS min_first_session_ping_run_date,
+      CASE
+        mozfun.norm.get_earliest_value(
+          [
+            (
+              STRUCT(CAST(first_session.adjust_network AS STRING), first_session.first_run_datetime)
+            ),
+            (STRUCT(CAST(metrics.adjust_network AS STRING), metrics.min_submission_datetime))
+          ]
+        )
+      WHEN
+        STRUCT(first_session.adjust_network, first_session.first_run_datetime)
+      THEN
+        'first_session'
+      WHEN
+        STRUCT(metrics.adjust_network, metrics.min_submission_datetime)
+      THEN
+        'metrics'
+      ELSE
+        NULL
+      END
+      AS adjust_network__source_ping,
+      'metrics' AS install_source__source_ping,
+      mozfun.norm.get_earliest_value(
+        [
+          (STRUCT(CAST(first_session.adjust_network AS STRING), first_session.first_run_datetime)),
+          (STRUCT(CAST(metrics.adjust_network AS STRING), metrics.min_submission_datetime))
+        ]
+      ).earliest_date AS adjust_network__source_ping_datetime,
+      CASE
+      WHEN
+        metrics.install_source IS NOT NULL
+      THEN
+        metrics.min_submission_datetime
+      ELSE
+        NULL
+      END
+      AS install_source__source_ping_datetime
+    ) AS metadata
+  FROM
+    first_seen
+  LEFT JOIN
+    first_session_ping first_session
+  USING
+    (client_id)
+  LEFT JOIN
+    metrics_ping AS metrics
+  USING
+    (client_id)
+)
+SELECT
+  *
+FROM
+  existing_with_pings_new_data
+UNION ALL
+SELECT
+  first_seen_with_pings_new_data.*
+FROM
+  first_seen_with_pings_new_data
+LEFT JOIN
+  existing_with_pings_new_data
+USING
+  (client_id)
+WHERE
+  existing_with_pings_new_data.client_id IS NULL
```

Comment:
I really think this should be an outer join. That is simpler, easier to understand, and matches our existing patterns (e.g. `clients_last_seen`).
```
WITH current AS (
  SELECT $fields
  FROM
    existing
  LEFT JOIN
    first_session_ping first_session
  USING
    (client_id)
  LEFT JOIN
    metrics_ping AS metrics
  USING
    (client_id)
  LEFT JOIN
    first_seen
)

SELECT
  IF(previous.client_id IS NULL, previous.*, current.*) REPLACE (
    LEAST(previous.first_seen_date, current.first_seen_date) AS first_seen_date,
    COALESCE(previous.adjust_campaign, current.adjust_campaign) AS adjust_campaign,
    COALESCE(previous.adjust_network, current.adjust_network) AS adjust_network,
    ...other fields...
  )
FROM
  current
FULL OUTER JOIN
  previous
  USING (client_id)
  ```

Reply:
Great, let's align with the current implementations.